pkg/devfile/devfile.go

```diff
@@ -90,6 +95,74 @@ func ConvertApplicationToDevfile(hasApp appstudiov1alpha1.Application, gitOpsRep
 	return devfileData, nil
 }
 
+func CreateDevfileForDockerfileBuild(uri, context string) (data.DevfileData, error) {
+	devfileVersion := string(data.APISchemaVersion220)
+	devfileData, err := data.NewDevfileData(devfileVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	devfileData.SetSchemaVersion(devfileVersion)
+
+	devfileData.SetMetadata(devfile.DevfileMetadata{
+		Name:        "dockerfile-component",
+		Description: "Basic Devfile for a Dockerfile Component",
+	})
+
+	components := []v1alpha2.Component{
+		{
+			Name: "dockerfile-build",
+			ComponentUnion: v1alpha2.ComponentUnion{
+				Image: &v1alpha2.ImageComponent{
+					Image: v1alpha2.Image{
+						ImageUnion: v1alpha2.ImageUnion{
+							Dockerfile: &v1alpha2.DockerfileImage{
+								DockerfileSrc: v1alpha2.DockerfileSrc{
+									Uri: uri,
+								},
+								Dockerfile: v1alpha2.Dockerfile{
+									BuildContext: context,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "container",
```

Comment:
Why is a container component needed here if it's no-op?

Reply:
because when we create a HAS `Component`, we are allowed to customize `Deployment` configs like ENV, Memory, CPU, etc and for `Service` like ports. 

Currently we do this via a Component CR right? And we store these info in the Devfile `Container` component. So this is why a no-op container component is required for handling Component CR updates.

Wouldnt the Image Src require the same thing in PR https://github.com/redhat-appstudio/application-service/pull/92?

Reply:
Okay, so it's the same reason as in my PR - to be able to store things like env, memory, etc config changes